src/main/java/hudson/remoting/Checksum.java

```diff
@@ -74,8 +74,10 @@ static Checksum forFile(File file) throws IOException {
     static Checksum forURL(URL url) throws IOException {
         try {
             MessageDigest md = MessageDigest.getInstance(JarLoaderImpl.DIGEST_ALGORITHM);
-            Util.copy(url.openStream(), new DigestOutputStream(new NullOutputStream(), md));
-            return new Checksum(md.digest(), md.getDigestLength() / 8);
+            try(OutputStream ostream = new DigestOutputStream(new NullOutputStream(), md)) {
```

Comment:
Unless `DigestOutputStream.<init>` throws an exception. That is why as a rule you should declare every `Closeable` object you are constructing as a separate variable in the block.